common/app/routes/Challenges/views/Modern/Show.jsx

```diff
@@ -24,12 +24,17 @@ const propTypes = {
 
 const mapStateToProps = createSelector(
   filesSelector,
-  files => ({
-    nameToFileKey: _.reduce(files, (map, file) => {
-      map[file.name] = file.key;
-      return map;
-    }, {})
-  })
+  files => {
+    if (Object.keys(files).length === 1) {
+      return { nameToFileKey: { Editor: files[Object.keys(files)[0]].key }};
```

Comment:
`files[Object.keys(files)[0]].key` This logic is quite complex. We can simplify this by using lodash.get https://lodash.com/docs/4.17.4#get

Reply:
@BerkeleyFalse if we know what the key of the only file then yes this is a little complex. I was assuming we don't know the name of the only file. Looking at the files object, i think this can be simplified to `Object.keys(files)[0]` but i am not sure if we can guarantee that if there is only one file it is always `indexjsx`.

Reply:
Ah yes, forgotten about that. It will get more complex with lodash but it will at least be declarative.

We need to get the values, then the first, then the property of key,

`_.flow(_.values, _.first, _.property('key'))({ foo: { key: 'bar' } }) // => 'bar'`

Looks like you are using this multiple times so it can be hoisted up into it's own function.

Reply:
Sounds good! Pushing up a change.